### ISSUE TITLE
semantic: incorrect semantic check for has same name `LabeledStatement` nested in `LabeledStatement` 

### ISSUE THREAD

**Issue by Dunqing** [oxc-project/oxc#5036]
## Reproduction: 

This bug was found in https://github.com/oxc-project/monitor-oxc/actions/runs/10482911458/job/29034791059

### Minimal reproduction
[oxc playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICzgICAgICAgICqk0bEZ9wjbyHIrGpkeLan7ycfX8yDD2f99XdYoe9ohorduk5Wq39%2FbTrAgA%3D%3D)

```js
TOP: {
  L: console.log(1);
  L: console.log(2);
};
```

The above code reports the following errors in the playground
```shell
❗ Identifier `L` has already been declared
❗ Identifier `L` has already been declared
```

### Expected result

There are no semantic errors. I've run the code in Chrome devtools and that works well

**Comment by Boshen** [comment]
We first need to change the diagnostic here 

https://github.com/oxc-project/oxc/blob/a4247e935317c59811c0b1338eefc2d8294b60a1/crates/oxc_semantic/src/checker/javascript.rs#L647

And then figure out what's wrong in the label builder https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/src/label.rs
